### PR TITLE
Create scconv.py

### PR DESCRIPTION
**Pull request by njmdin12**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Add a new SCConv residual convolution module to ultralytics/nn/modules 🎉

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @username, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

Additional notes for this PR:
- Consider adding shape-safe residual handling (e.g., a 1x1 projection) if `stride != 1` or `in_channels != out_channels` to prevent runtime errors.
- Please add an import in `ultralytics/nn/modules/__init__.py` so the new `SCConv` is discoverable.
- A brief docstring and minimal unit test would help maintainability and CI coverage.

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). This is an automated message, and an Ultralytics engineer will assist you shortly. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/nn/modules/scconv.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        self.conv2 = nn.Conv2d(out_channels, out_channels, kernel_size, stride=1, padding=1, groups=groups, bias=False)
+        self.bn2 = nn.BatchNorm2d(out_channels)
+        self.relu = nn.ReLU(inplace=True)
+
+    def forward(self, x):
+        identity = x
+        out = self.conv1(x)
+        out = self.bn1(out)
+        out = self.relu(out)
+        out = self.conv2(out)
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/modules/scconv.py]: ⚠️ **HIGH**: Adding `identity` back to the output assumes `in_channels == out_channels` and `stride == 1`. As written we expose `out_channels` and `stride` as configurable, so a caller can easily trigger a runtime shape mismatch when `stride != 1` or the channel counts differ. Please either constrain the constructor arguments (and document it) or add a projection/pooling path so the residual branch always matches the main branch.

File: ultralytics/nn/modules/scconv.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class SCConv(nn.Module):
+    def __init__(self, in_channels, out_channels, kernel_size=3, stride=1, padding=1, groups=1):
+        super(SCConv, self).__init__()
+        self.conv1 = nn.Conv2d(in_channels, out_channels, kernel_size, stride, padding, groups=groups, bias=False)
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/modules/scconv.py:7]: 💡 **MEDIUM**: `self.conv2` hardcodes `padding=1`, but we expose `kernel_size` as a parameter. For `kernel_size != 3` this will break spatial alignment; e.g., `kernel_size=1` still pads by 1. Consider reusing the `padding` argument or computing padding dynamically to keep the feature map shape consistent.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The residual branch currently breaks when `stride` or channel counts differ, so please align the skip path with the main path. Also, adjust `conv2` to honor custom `kernel_size` values instead of hardcoding `padding=1`.

💬 Posted 2 inline comments
